### PR TITLE
chore: add yamllint rules for braces

### PR DESCRIPTION
**Pull request by h-ohta**
according to https://prettier.io/docs/en/options.html#bracket-spacing, Prettier has the bracket rule and printing spaces between brackets is default.
So, This PR adds some yamllint rules which define max spaces inside brackets and braces .

### CODE REVIEW

**Comment by kenji-miyake** [comment]
Thank you!

The default rule of `yamllint` is here.
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.braces
https://yamllint.readthedocs.io/en/stable/configuration.html#default-configuration

Since `relaxed` configuration uses `max-spaces-inside: 1` and `Prettier` automatically formats so by default, I think it's good.

File: .yamllint.yaml [hunk 1]
```diff
@@ -5,6 +5,12 @@ ignore: |
   *.param.yaml
 
 rules:
+  braces:
+    level: error
+    max-spaces-inside: 1
+  brackets:
+    level: error
+    max-spaces-inside: 1
```
Comment (kenji-miyake) [inline .yamllint.yaml]: I think this is for `[]`, not `{}`.
Could you show me the reference for this?

Currently, Prettier formats files with no space.
https://github.com/autowarefoundation/autoware/blob/7c4c396fce51ee2a0c50e3575c77e870e22bebe6/.pre-commit-config.yaml#L26

Comment (h-ohta) [reply .yamllint.yaml]: I thought `Bracket Spacing` applied to both `{}` and  `[]`.

According to prettier documentation, `Bracket Spacing` only applies to `{}`. 
On the other hands, yamllint documentation says that `braces` rule only applies to `{}` and `brackets` rule only applies to `[]` .

This is my misunderstanding.
I suppose that I can delete `brackets` rule.

File: .yamllint.yaml [hunk 2]
```diff
@@ -5,6 +5,9 @@ ignore: |
   *.param.yaml
 
 rules:
+  braces:
+    level: error
+    max-spaces-inside: 1
```
Comment (kenji-miyake) [inline .yamllint.yaml]: Can I add a comment?

```suggestion
    max-spaces-inside: 1 # To autoformat with Prettier
```

File: .yamllint.yaml [hunk 3]
```diff
@@ -5,6 +5,9 @@ ignore: |
   *.param.yaml
 
 rules:
+  braces:
+    level: error
+    max-spaces-inside: 1 # To autoformat with Prettier
```
Comment (kenji-miyake) [inline .yamllint.yaml]: Ah, I'm sorry `autoformat` was not in the dictionary.
image

Which do you think is better adding to the dictionary or rephrasing it?

Comment (kenji-miyake) [reply .yamllint.yaml]: Is this enough?

```suggestion
    max-spaces-inside: 1 # To format with Prettier
```

Comment (h-ohta) [reply .yamllint.yaml]: I think rephrasing is better.
How about `# To format with Prettier automatically` ?

Comment (kenji-miyake) [reply .yamllint.yaml]: Hmm, I feel `auto` was obvious and can be removed.

Comment (h-ohta) [reply .yamllint.yaml]: I think so.

**Review by kenji-miyake — APPROVED** [review]
Thank you!

**Comment by h-ohta** [comment]
I fix DCO. wait a moment...

**Comment by kenji-miyake** [comment]
@h-ohta I did it from the browser.

**Comment by kenji-miyake** [comment]
Since it's not `feat`, I changed the title to `chore`.

**Pull request merged**